### PR TITLE
gcc7 compilation warning(s) fix(es) in RecoGamma & RecoMuon

### PR DESCRIPTION
**Pull request by mrodozov**
Fix for compilation warning(s) when compiling with gcc7 and more flags listed here:
https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc6_amd64_gcc700/CMSSW_9_3_X_2017-07-24-2300/RecoMuon/MuonIsolation
https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc6_amd64_gcc700/CMSSW_9_3_X_2017-07-24-2300/RecoEgamma/EgammaIsolationAlgos
https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc6_amd64_gcc700/CMSSW_9_3_X_2017-07-24-2300/RecoMuon/TrackingTools

### CODE REVIEW

File: RecoMuon/TrackingTools/src/MuonTrackLoader.cc [hunk 1]
```diff
@@ -202,7 +202,7 @@ MuonTrackLoader::loadTracks(const TrajectoryContainer& trajectories,
   reco::TrackRef::key_type trackUpdatedIndex = 0;
   
   reco::TrackExtraRef::key_type trackExtraIndex = 0;
-  TrackingRecHitRef::key_type recHitsIndex = 0;
+  //TrackingRecHitRef::key_type recHitsIndex = 0;
```
Comment (perrotta) [inline RecoMuon/TrackingTools/src/MuonTrackLoader.cc]: Please remove this commented out code

File: RecoMuon/TrackingTools/src/MuonTrackLoader.cc [hunk 2]
```diff
@@ -544,7 +544,7 @@ MuonTrackLoader::loadTracks(const TrajectoryContainer& trajectories,
   //  reco::TrackRef::key_type trackUpdatedIndex = 0;
```
Comment (perrotta) [inline RecoMuon/TrackingTools/src/MuonTrackLoader.cc]: Please remove this commented out code

File: RecoMuon/TrackingTools/src/MuonTrackLoader.cc [hunk 3]
```diff
@@ -544,7 +544,7 @@ MuonTrackLoader::loadTracks(const TrajectoryContainer& trajectories,
...
   
   reco::TrackExtraRef::key_type trackExtraIndex = 0;
-  TrackingRecHitRef::key_type recHitsIndex = 0;
+  //TrackingRecHitRef::key_type recHitsIndex = 0;
```
Comment (perrotta) [inline RecoMuon/TrackingTools/src/MuonTrackLoader.cc]: Please remove this commented out code

**Comment by davidlange6** [comment]
please test

**Comment by perrotta** [comment]
@mrodozov : do you plan to remove the commented out lines, as suggested in https://github.com/cms-sw/cmssw/pull/19904#pullrequestreview-52209490 ? 

Otherwise please add some further comment in the code which explains why they are needed

**Comment by mrodozov** [comment]
please test

**Comment by perrotta** [comment]
+1
Technical
No changes expected, no changes observed in tests
It does not involve recoEGamma any more

**Comment by davidlange6** [comment]
+1

**Pull request merged**